packages/editor/src/components/header/index.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { useViewportMatch } from '@wordpress/compose';
+import { __unstableMotion as motion } from '@wordpress/components';
+import { store as preferencesStore } from '@wordpress/preferences';
+import { useState } from '@wordpress/element';
+import { PinnedItems } from '@wordpress/interface';
+
+/**
+ * Internal dependencies
+ */
+import CollapsableBlockToolbar from '../collapsible-block-toolbar';
+import DocumentBar from '../document-bar';
+import DocumentTools from '../document-tools';
+import MoreMenu from '../more-menu';
+import PostPreviewButton from '../post-preview-button';
+import PostPublishButtonOrToggle from '../post-publish-button/post-publish-button-or-toggle';
+import PostSavedState from '../post-saved-state';
+import PostTypeSupportCheck from '../post-type-support-check';
+import PostViewLink from '../post-view-link';
+import PreviewDropdown from '../preview-dropdown';
+import { store as editorStore } from '../../store';
+
+const slideY = {
+	hidden: { y: '-50px' },
+	distractionFreeInactive: { y: 0 },
+	hover: { y: 0, transition: { type: 'tween', delay: 0.2 } },
+};
+
+function Header( {
+	customSaveButton,
+	forceIsDirty,
+	setEntitiesSavedStatesCallback,
+	children,
+} ) {
+	const isWideViewport = useViewportMatch( 'large' );
+	const isLargeViewport = useViewportMatch( 'medium' );
+	const {
+		isTextEditor,
+		isPublishSidebarOpened,
+		showIconLabels,
+		hasFixedToolbar,
+	} = useSelect( ( select ) => {
+		const { get: getPreference } = select( preferencesStore );
+		const {
+			getEditorMode,
+			isPublishSidebarOpened: _isPublishSidebarOpened,
+		} = select( editorStore );
+
+		return {
+			isTextEditor: getEditorMode() === 'text',
+			isPublishSidebarOpened: _isPublishSidebarOpened(),
+			showIconLabels: getPreference( 'core', 'showIconLabels' ),
+			hasFixedToolbar: getPreference( 'core', 'fixedToolbar' ),
+		};
+	}, [] );
+
+	const hasTopToolbar = isLargeViewport && hasFixedToolbar;
+
+	const [ isBlockToolsCollapsed, setIsBlockToolsCollapsed ] =
+		useState( true );
+
+	// The edit-post-header classname is only kept for backward compatibilty
+	// as some plugins might be relying on its presence.
+	return (
+		<div className="editor-header edit-post-header">
+			{ children }
+			<motion.div
+				variants={ slideY }
+				transition={ { type: 'tween', delay: 0.8 } }
+				className="editor-header__toolbar"
```

Comment:
The site editor hides the `editor-header__toolbar` div (previously `edit-site-header-edit-mode__start`) when rendering Style Book or Revision canvases.

They also render custom `DocumentBar`; though it's "broken" on the trunk, we might want to retain the previous behavior.

Related: https://github.com/WordPress/gutenberg/issues/51667 and https://github.com/WordPress/gutenberg/issues/54514.

| trunk | this branch |
|--------|--------|
| ![CleanShot 2024-05-01 at 15 20 42](https://github.com/WordPress/gutenberg/assets/240569/b2d546ae-3198-4885-a055-c4be24d63415) | ![CleanShot 2024-05-01 at 15 21 27](https://github.com/WordPress/gutenberg/assets/240569/64bc807a-804c-4bcc-951f-4c7256b3a9bb) |

Reply:
Yes, I mention this on the PR description, that notion of "active view" in the site editor is not perfect. Not sure yet what I want to do tbh.

Reply:
Here are my thoughts:

 - Undo and redo are useful
 - Inserter and list view are not
 - The DocumentBar is useful because it seems like we're saying that the "active entity" is the global styles entity, which means, should we consider these as "focus mode" to the global styles entity instead of these adhoc tweaks. In other words should we change the postType/postId passed to the EditorProvider instead in this case.

Reply:
Sorry, I missed that.

* We could use `disableBlockTools` for those views. Similar to the code editor.
* The `DocumentBar` - follow, @rich's suggestions from #51667 and #54514.

Reply:
I've added a couple props to the Header component to address this. It's not a perfect solution but it allows us to move forward for now until we reach a point where we could potentially make the global styles an editable entity like all the others in the editor package.